### DISCUSSION TITLE
Github page is deploy but not work

### DISCUSSION THREAD

**Discussion by Zutyosh** [community/community#67634] [Actions] [answered]
### Select Topic Area

Bug

### Body

Hello,
Today, I deployed my website built to Go with the Hugo framework via GitHub Page, after a few configuration errors, I managed to get it working and everything worked great!

But for some obscure reason, after I added a favicon and changed the page title in config.toml the deployment still went ahead without a hitch, but now my site's web page is displaying a 404 error, even though these settings haven't changed. I don't understand, does anyone have an answer?

My project : https://github.com/TheFlomax/trapmesenpai.wtf

**Comment by seyLu** [comment, marked as answer]
#### Hi @TheFlomax,

Github pages looks for `index.html` at the root dir. In this case since you are building and deploying from a workflow and not a branch, github pages will look for `index.html` at the root dir of the `artifact` you uploaded.

For example, from `your site's artifact`:

<img alt="image" src="https://github.com/community/community/assets/98249191/6be52303-f28a-4c60-9376-5838ee103f59">
<table>
<td>
<img alt="image" src="https://github.com/community/community/assets/98249191/eec2e490-556b-496f-bc6a-fad797e21618">
</td>
<td>
<img alt="image" src="https://github.com/community/community/assets/98249191/504047c5-55ab-4a18-b988-09629838c9ca">
</td>
</table> 

<br>

And according to Hugo docs, if the build is successful, it should recreate your site to a public dir with this structure (notice the `index.html` at the root dir):
![image](https://github.com/community/community/assets/98249191/731a5be7-63ec-4547-a380-5767b5524979)

<br>

I've also noticed that you've commented out `baseUrl` in `config.toml`:
![image](https://github.com/community/community/assets/98249191/b20282b1-3d3b-42c9-88f2-ff3f346f7a16)

<br>  

which messes up this part of the workflow:
![image](https://github.com/community/community/assets/98249191/2fafa41c-46d8-4c02-aabe-8fc9009faca0)

<br>

I've went and [deployed a copy of your site](https://seylu.github.io/trapmesenpai.wtf/). Things I've changed:
- configure `baseUrl` in `config.toml` to point to site
- change author img path from relative to absolute

```toml
baseURL = "https://seyLu.github.io/trapmesenpai.wtf/"
...

[author]
   name = "trapmesenpai.wtf"
   headline = "Bienvenu sur le portail des services auto-héberger de trapmesenpai.wtf"
   image = "img/astolfo.webp"
```

<br>

You can read more on:
- [Hugo's directory structure explained](https://www.jakewiesler.com/blog/hugo-directory-structure)
- [Hugo docs: directory structure](https://gohugo.io/getting-started/directory-structure/)
- [Hugo docs: Build your site](https://gohugo.io/getting-started/usage/#build-your-site)

<br>

PS: [astolfo did win best girl 2017](https://twitter.com/lashin_akiba2/status/900569697422237699)
<img width="500" alt="image" src="https://github.com/community/community/assets/98249191/0248732d-9779-4b70-ba84-3d65f01e4494">

**Reply by Zutyosh** [reply]
Hello,
Thank you so much for taking the time to provide such a detailed response!
Thanks to you, I've managed to deploy my website correctly.

I hope to see you again on my future Firefish instance (social.trapmesenpai.wtf) where you're welcome.

P.S. At the same time, it's well known that boys make better girls uwu
![image](https://github.com/community/community/assets/69845574/cadb1f23-52f6-4336-b538-fdd06e07bddf)
